### PR TITLE
Fix range of commits displayed when merge commits are present

### PR DESCRIPTION
**Pull request by PawelLipski**
Wrt. the original issue:

image

### CODE REVIEW

File: gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/BaseGitCoreBranch.java [hunk 1]
```diff
@@ -74,12 +74,19 @@ public synchronized GitCoreCommit getPointedCommit() throws GitCoreException {
   @Override
   public List<IGitCoreCommit> deriveCommitsUntil(IGitCoreCommit upToCommit) throws GitCoreException {
     LOG.debug(() -> "Entering: branch = '${getFullName()}', upToCommit = ${upToCommit}");
+    @Unique RevWalk walk = getTopoRevWalkFromPointedCommit();
+    try {
+      walk.markUninteresting(walk.parseCommit(repo.gitCoreCommitToObjectId(upToCommit)));
```
Comment (PawelLipski) [inline gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/BaseGitCoreBranch.java:79]:
```java
	/**
	 * Include {@link RevFlag#UNINTERESTING} boundary commits after all others.
	 * In {@link ObjectWalk}, objects associated with such commits (trees,
	 * blobs), and all other objects marked explicitly as UNINTERESTING are also
	 * included.
	 * <p>
	 * A boundary commit is a UNINTERESTING parent of an interesting commit that
	 * was previously output.
	 */
	BOUNDARY;
```

File: config/checkstyle/tree_walker_module_directives.xml [hunk 2]
```diff
@@ -121,12 +121,6 @@
     <property name="format" value="\bpeek\b"/>
     <property name="message" value="Usage of the prohibited peek method"/>
 </module>
-<module name="Regexp">
-    <property name="illegalPattern" value="true"/>
-    <property name="ignoreComments" value="true"/>
-    <property name="format" value="\bSneakyThrows\b"/>
-    <property name="message" value="String interpolation does not work in methods marked with lombok.SneakyThrows, use lombok.Lombok#sneakyThrow instead"/>
```
Comment (mkondratek) [inline config/checkstyle/tree_walker_module_directives.xml:128]: Is it no longer true?

File: gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/BaseGitCoreBranch.java [hunk 3]
```diff
@@ -74,12 +74,19 @@ public synchronized GitCoreCommit getPointedCommit() throws GitCoreException {
...
+    }
+    walk.sort(RevSort.BOUNDARY);
 
     LOG.debug("Starting revwalk");
-    return Iterator.ofAll(getTopoRevWalkFromPointedCommit())
+    return Iterator.ofAll(walk.iterator())
         .takeUntil(revCommit -> revCommit.getId().getName().equals(upToCommit.getHash().getHashString()))
         .map(revCommit -> {
-          LOG.debug(() -> revCommit.getId().getName());
+          LOG.debug(() -> "* " + revCommit.getId().getName());
```
Comment (mkondratek) [inline gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/BaseGitCoreBranch.java:89]: string interpolations here?

File: config/checkstyle/tree_walker_module_directives.xml (hunk 2 above)
Comment (PawelLipski) [reply config/checkstyle/tree_walker_module_directives.xml:128]: It was likely never true in the first place, I've only ever observed that in `binding` module and thought that's the fault of `@SneakyThrows`, but the actual reason was/is sth different... still dunno what specifically

File: gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/BaseGitCoreBranch.java (hunk 3 above)
Comment (PawelLipski) [reply gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/BaseGitCoreBranch.java:89]: IMO more readable without here :) string interpolations IMO make less sense when there's just one value to be interpolated at the beginning or end of a short string

**Pull request merged**